huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go

```diff
@@ -0,0 +1,304 @@
+	_, err = updateIdentityCenterUserClient.Request("PUT", updateIdentityCenterUserPath,
+		&updateIdentityCenterUserOpt)
+	if err != nil {
+		return diag.Errorf("error updating IdentityCenter: %s", err)
+	}
+	return resourceIdentityCenterUserRead(ctx, d, meta)
+}
+
+func buildUpdateIdentityCenterUserBodyParams(d *schema.ResourceData) map[string]interface{} {
+	operations := make([]map[string]interface{}, 0)
+	if d.HasChanges("family_name", "given_name") {
+		updateValue := map[string]interface{}{
+			"family_name": utils.ValueIngoreEmpty(d.Get("family_name")),
+			"given_name":  utils.ValueIngoreEmpty(d.Get("given_name")),
+		}
+		updateValueJson, _ := json.Marshal(updateValue)
+		operations = append(operations, map[string]interface{}{
+			"attribute_path":  "name",
+			"attribute_value": string(updateValueJson),
+		})
+	}
+	if d.HasChange("display_name") {
+		operations = append(operations, map[string]interface{}{
+			"attribute_path":  "display_name",
+			"attribute_value": d.Get("display_name"),
+		})
+	}
+	if d.HasChange("email") {
+		updateValue := []map[string]interface{}{{
+			"primary": true,
+			"type":    "Work",
+			"value":   utils.ValueIngoreEmpty(d.Get("email")),
+		}}
+		updateValueJson, _ := json.Marshal(updateValue)
+		operations = append(operations, map[string]interface{}{
+			"attribute_path":  "emails",
+			"attribute_value": string(updateValueJson),
+		})
+	}
+	return map[string]interface{}{"operations": operations}
+}
+
+func resourceIdentityCenterUserDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteIdentityCenterUser: Delete Identity Center user
+	var (
+		deleteIdentityCenterUserHttpUrl = "v1/identity-stores/{identity_store_id}/users/{user_id}"
+		deleteIdentityCenterUserProduct = "identitystore"
+	)
+	deleteIdentityCenterUserClient, err := cfg.NewServiceClient(deleteIdentityCenterUserProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating IdentityCenter Client: %s", err)
+	}
+
+	deleteIdentityCenterUserPath := deleteIdentityCenterUserClient.Endpoint + deleteIdentityCenterUserHttpUrl
+	deleteIdentityCenterUserPath = strings.ReplaceAll(deleteIdentityCenterUserPath, "{identity_store_id}",
+		fmt.Sprintf("%v", d.Get("identity_store_id")))
+	deleteIdentityCenterUserPath = strings.ReplaceAll(deleteIdentityCenterUserPath, "{user_id}", d.Id())
+
+	deleteIdentityCenterUserOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	_, err = deleteIdentityCenterUserClient.Request("DELETE", deleteIdentityCenterUserPath,
+		&deleteIdentityCenterUserOpt)
+	if err != nil {
+		return diag.Errorf("error deleting IdentityCenterUser: %s", err)
+	}
+
+	return nil
+}
+
+func resourceIdentityCenterUserImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
```

Comment:
use *strings.Split* will be more suitable.

Reply:
fixed